frontend/herdForm.tsx

```diff
@@ -51,9 +52,8 @@ export function HerdForm(props: {id: string | number | undefined}) {
     }
   }, [props])
 
-  const setFormField = (label: string, value: string | number) => {
-    herd[label] = value;
-    setHerd(herd);
+  const setFormField = <K extends keyof Herd>(label: K, value: Herd[K]) => {
+    herd && {...herd, label: value}
```

Comment:
You lost `setHerd` :fearful:

Reply:
And the brackets in `[label]`!

`setHerd({...herd, [label]: value})`

Reply:
haha - yeah, that was dumb ^^